### PR TITLE
CI: Add mkosi boot tests

### LINKED ISSUE #17828: test_exec_dynamicuser fails when building mkosi image
While trying to get #17787 working, I discovered a test failure when building the Arch or Fedora mkosi image. (Might also need to add rpm to the list of packages before that for Fedora, that's another bug)

To reproduce: `mkosi --default .mkosi/mkosi.fedora`

```
--- command ---
22:45:22 PATH='/root/build:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin' SYSTEMD_KBD_MODEL_MAP='/root/src/src/locale/kbd-model-map' SYSTEMD_LANGUAGE_FALLBACK_MAP='/root/src/src/locale/language-fallback-map' /root/build/test-execute
--- Listing only the last 100 lines from a long log. ---
exec-dynamicuser-statedir.service: Forked touch as 2485
exec-dynamicuser-statedir.service: User lookup succeeded: uid=65325 gid=65325
Received SIGCHLD from PID 2485 (touch).
Child 2485 (touch) died (code=exited, status=0/SUCCESS)
exec-dynamicuser-statedir.service: Failed to read oom_kill field of memory.events cgroup attribute: No such file or directory
exec-dynamicuser-statedir.service: Child 2485 belongs to exec-dynamicuser-statedir.service.
exec-dynamicuser-statedir.service: Main process exited, code=exited, status=0/SUCCESS
exec-dynamicuser-statedir.service: Running next main command for state start.
exec-dynamicuser-statedir.service: Passing 0 fds to service
exec-dynamicuser-statedir.service: About to execute touch /var/lib/quux/pief/yayyay
exec-dynamicuser-statedir.service: Forked touch as 2486
exec-dynamicuser-statedir.service: User lookup succeeded: uid=65325 gid=65325
Received SIGCHLD from PID 2486 (touch).
Child 2486 (touch) died (code=exited, status=0/SUCCESS)
exec-dynamicuser-statedir.service: Failed to read oom_kill field of memory.events cgroup attribute: No such file or directory
exec-dynamicuser-statedir.service: Child 2486 belongs to exec-dynamicuser-statedir.service.
exec-dynamicuser-statedir.service: Main process exited, code=exited, status=0/SUCCESS
exec-dynamicuser-statedir.service: Running next main command for state start.
exec-dynamicuser-statedir.service: Passing 0 fds to service
exec-dynamicuser-statedir.service: About to execute test -f /var/lib/waldo/yay
exec-dynamicuser-statedir.service: Forked test as 2487
exec-dynamicuser-statedir.service: User lookup succeeded: uid=65325 gid=65325
Received SIGCHLD from PID 2487 (test).
Child 2487 (test) died (code=exited, status=0/SUCCESS)
exec-dynamicuser-statedir.service: Failed to read oom_kill field of memory.events cgroup attribute: No such file or directory
exec-dynamicuser-statedir.service: Child 2487 belongs to exec-dynamicuser-statedir.service.
exec-dynamicuser-statedir.service: Main process exited, code=exited, status=0/SUCCESS
exec-dynamicuser-statedir.service: Running next main command for state start.
exec-dynamicuser-statedir.service: Passing 0 fds to service
exec-dynamicuser-statedir.service: About to execute test -f /var/lib/quux/pief/yayyay
exec-dynamicuser-statedir.service: Forked test as 2488
exec-dynamicuser-statedir.service: User lookup succeeded: uid=65325 gid=65325
Received SIGCHLD from PID 2488 (test).
Child 2488 (test) died (code=exited, status=0/SUCCESS)
exec-dynamicuser-statedir.service: Failed to read oom_kill field of memory.events cgroup attribute: No such file or directory
exec-dynamicuser-statedir.service: Child 2488 belongs to exec-dynamicuser-statedir.service.
exec-dynamicuser-statedir.service: Main process exited, code=exited, status=0/SUCCESS
exec-dynamicuser-statedir.service: Running next main command for state start.
exec-dynamicuser-statedir.service: Passing 0 fds to service
exec-dynamicuser-statedir.service: About to execute test -f /var/lib/private/waldo/yay
exec-dynamicuser-statedir.service: Forked test as 2489
exec-dynamicuser-statedir.service: User lookup succeeded: uid=65325 gid=65325
Received SIGCHLD from PID 2489 (test).
Child 2489 (test) died (code=exited, status=0/SUCCESS)
exec-dynamicuser-statedir.service: Failed to read oom_kill field of memory.events cgroup attribute: No such file or directory
exec-dynamicuser-statedir.service: Child 2489 belongs to exec-dynamicuser-statedir.service.
exec-dynamicuser-statedir.service: Main process exited, code=exited, status=0/SUCCESS
exec-dynamicuser-statedir.service: Running next main command for state start.
exec-dynamicuser-statedir.service: Passing 0 fds to service
exec-dynamicuser-statedir.service: About to execute test -f /var/lib/private/quux/pief/yayyay
exec-dynamicuser-statedir.service: Forked test as 2490
exec-dynamicuser-statedir.service: User lookup succeeded: uid=65325 gid=65325
Received SIGCHLD from PID 2490 (test).
Child 2490 (test) died (code=exited, status=0/SUCCESS)
exec-dynamicuser-statedir.service: Failed to read oom_kill field of memory.events cgroup attribute: No such file or directory
exec-dynamicuser-statedir.service: Child 2490 belongs to exec-dynamicuser-statedir.service.
exec-dynamicuser-statedir.service: Main process exited, code=exited, status=0/SUCCESS
exec-dynamicuser-statedir.service: Running next main command for state start.
exec-dynamicuser-statedir.service: Passing 0 fds to service
exec-dynamicuser-statedir.service: About to execute sh -x -c 'test "$$STATE_DIRECTORY" = "/var/lib/waldo:/var/lib/quux/pief"'
exec-dynamicuser-statedir.service: Forked sh as 2491
exec-dynamicuser-statedir.service: User lookup succeeded: uid=65325 gid=65325
Received SIGCHLD from PID 2491 (sh).
Child 2491 (sh) died (code=exited, status=0/SUCCESS)
exec-dynamicuser-statedir.service: Failed to read oom_kill field of memory.events cgroup attribute: No such file or directory
exec-dynamicuser-statedir.service: Child 2491 belongs to exec-dynamicuser-statedir.service.
exec-dynamicuser-statedir.service: Main process exited, code=exited, status=0/SUCCESS
exec-dynamicuser-statedir.service: Running next main command for state start.
exec-dynamicuser-statedir.service: Passing 0 fds to service
exec-dynamicuser-statedir.service: About to execute sh -x -c 'echo $$(find / \( -path /var/tmp -o -path /tmp -o -path /proc -o -path /dev/mqueue -o -path /dev/shm -o -path /sys/fs/bpf -o -path /dev/.lxc \) -prune -o -type d -writable -print 2>/dev/null | sort -u | tr -d "\\n")'
exec-dynamicuser-statedir.service: Forked sh as 2492
exec-dynamicuser-statedir.service: User lookup succeeded: uid=65325 gid=65325
Received SIGCHLD from PID 2492 (sh).
Child 2492 (sh) died (code=exited, status=0/SUCCESS)
exec-dynamicuser-statedir.service: Failed to read oom_kill field of memory.events cgroup attribute: No such file or directory
exec-dynamicuser-statedir.service: Child 2492 belongs to exec-dynamicuser-statedir.service.
exec-dynamicuser-statedir.service: Main process exited, code=exited, status=0/SUCCESS
exec-dynamicuser-statedir.service: Running next main command for state start.
exec-dynamicuser-statedir.service: Passing 0 fds to service
exec-dynamicuser-statedir.service: About to execute sh -x -c 'test $$(find / \( -path /var/tmp -o -path /tmp -o -path /proc -o -path /dev/mqueue -o -path /dev/shm -o -path /sys/fs/bpf -o -path /dev/.lxc \) -prune -o -type d -writable -print 2>/dev/null | sort -u | tr -d "\\n") = /var/lib/private/quux/pief/var/lib/private/waldo'
exec-dynamicuser-statedir.service: Forked sh as 2497
exec-dynamicuser-statedir.service: User lookup succeeded: uid=65325 gid=65325
Received SIGCHLD from PID 2497 (sh).
Child 2497 (sh) died (code=exited, status=2/INVALIDARGUMENT)
exec-dynamicuser-statedir.service: Failed to read oom_kill field of memory.events cgroup attribute: No such file or directory
exec-dynamicuser-statedir.service: Child 2497 belongs to exec-dynamicuser-statedir.service.
exec-dynamicuser-statedir.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
exec-dynamicuser-statedir.service: Failed with result 'exit-code'.
exec-dynamicuser-statedir.service: Service will not restart (restart setting)
exec-dynamicuser-statedir.service: Changed start -> failed
exec-dynamicuser-statedir.service: Unit entered failed state.
Spawning thread to nuke /tmp/systemd-private-e1ae8c44766b4857be2f75126a960c72-exec-dynamicuser-statedir.service-nqVfji
Spawning thread to nuke /var/tmp/systemd-private-e1ae8c44766b4857be2f75126a960c72-exec-dynamicuser-statedir.service-74XwAg
UID 65325 is no longer referenced, cleaning up its IPC.
GID 65325 is no longer referenced, cleaning up its IPC.
Failed to connect to nscd socket: No such file or directory
Failed to connect to nscd socket: No such file or directory
src/test/test-execute.c:584:test_exec_dynamicuser: exec-dynamicuser-statedir.service: exit status 2, expected 0
-------
```

### PR DESCRIPTION
**Pull request by daandemeyer**
Using the new mkosi Github Action, we can add some simple boot tests
for the systemd mkosi configs. This makes sure these keep working
as expected.

Also contains some extra commits fixing various bugs in the mkosi configs. (Fixes #17828)

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
Opensuse has test failures (test-execute and test-seccomp) and Fedora has a dnf failure. Maybe @keszybz knows more about how to solve that one? We can remove opensuse if we don't have anyone to look into the test failures there. 

This basically copies the boot test from mkosi. With the new mkosi action, this seems like a pretty low-cost way to make sure the mkosi config images build and boot as expected. The python script is a simple pexpect script that logs into the booted container/vm and shuts it down again.

**Comment by yuwata** [comment]
It sounds dnf is not installed. No?

**Comment by DaanDeMeyer** [comment]
The setup-mkosi action builds dnf (and pacman) from source. It's already used in mkosi's CI to build all kinds of different distro images so I'm pretty confident the issue's not there. I don't encounter the error locally though. Maybe the dnf version built by setup-mkosi has a bug?

**Comment by DaanDeMeyer** [comment]
Cannot reproduce the fedora failure on Fedora 33 or Arch. Updating all mkosi deps built from source to the latest versions didn't help either. One lib we don't build from source is librpm itself which comes from Ubuntu's repositories and is currently at 4.14. I'll try and build the latest librpm from source as well, maybe that fixes the issue.

File: .github/workflows/test_mkosi_boot.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
+
```
Comment (bluca) [inline .github/workflows/test_mkosi_boot.py:3]: \# SPDX-License-Identifier: LGPL-2.1-or-later

**Comment by bluca** [comment]
> meson.build:904:0: ERROR: Pkg-config binary for machine MachineChoice.HOST not found. Giving up.

**Comment by DaanDeMeyer** [comment]
I could have sworn I fixed that already. Fix incoming

**Comment by bluca** [comment]
I imagine the current failure is due to https://github.com/systemd/systemd/pull/17846

**Comment by DaanDeMeyer** [comment]
Nope, that one only occurs with systemd-nspawn v247 which Ubuntu Focal doesn't have. This failure is 17828, I'm trying to figure that one out but not getting very far unfortunately.

**Comment by DaanDeMeyer** [comment]
I expected some deep and complex issue in library differences between distros causing this issue. Turns out the Fedora container simply didn't have find installed.

**Comment by DaanDeMeyer** [comment]
(Also removed opensuse test as we removed it from mkosi's github action as well due to too much overhead in getting their package manager building from source on Focal)

**Comment by DaanDeMeyer** [comment]
Fixes #17828

**Comment by bluca** [comment]
Github actions are all green, nothing else is affected so no need to wait for autopkgtest, merging

**Pull request merged**